### ISSUE TITLE
MatchError with akka.typed.AskPattern

### ISSUE THREAD

**Issue by durban** [akka/akka#19947]
If the typed ask pattern is used on an ActorRef which refers to a nonexistent actor, the AskPattern implementation throws a MatchError (see below; full code is here: https://github.com/durban/akka-typed-examples/blob/8625c5404d109b30a2a5fa0e85137baa7ed2e41f/src/main/scala/com/example/ask/TypedAsk.scala). The reason seems to be that a destructuring val definition tries to match a 3-tuple, but in this case a 2-tuple is returned.

```
[info] Running com.example.ask.TypedAsk 
[ERROR] [03/04/2016 10:34:10.155] [MySys-akka.actor.default-dispatcher-4] [akka.actor.LocalActorRefProvider(akka://MySys)] guardian failed, shutting down system
scala.MatchError: (Actor[akka://MySys/deadLetters],scala.concurrent.impl.Promise$KeptPromise@1603220d) (of class scala.Tuple2)                                
        at akka.typed.AskPattern$PromiseRef.<init>(Ask.scala:42)                                                                                              
        at akka.typed.AskPattern$PromiseRef$.apply(Ask.scala:60)                                                                                              
        at akka.typed.AskPattern$.ask(Ask.scala:64)                                                                                                           
        at akka.typed.AskPattern$Askable$.$qmark$extension(Ask.scala:38)                                                      
        at com.example.ask.TypedAsk$$anonfun$root$1$$anonfun$apply$2.apply(TypedAsk.scala:26)                                 
        at com.example.ask.TypedAsk$$anonfun$root$1$$anonfun$apply$2.apply(TypedAsk.scala:22)                                 
        at akka.typed.ScalaDSL$Static.message(ScalaDSL.scala:240)                                                             
        at akka.typed.ActorAdapter$$anonfun$receive$1.applyOrElse(Impl.scala:31)                                              
        at akka.actor.Actor$class.aroundReceive(Actor.scala:480)                                                              
        at akka.typed.ActorAdapter.aroundReceive(Impl.scala:17)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:526)
        at akka.actor.ActorCell.invoke(ActorCell.scala:495)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
        at akka.dispatch.Mailbox.run(Mailbox.scala:224)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)

[ERROR] [03/04/2016 10:34:10.160] [MySys-akka.actor.default-dispatcher-4] [akka://MySys/user] (Actor[akka://MySys/deadLetters],scala.concurrent.impl.Promise$KeptPromise@1603220d) (of class scala.Tuple2)
scala.MatchError: (Actor[akka://MySys/deadLetters],scala.concurrent.impl.Promise$KeptPromise@1603220d) (of class scala.Tuple2)
        at akka.typed.AskPattern$PromiseRef.<init>(Ask.scala:42)
        at akka.typed.AskPattern$PromiseRef$.apply(Ask.scala:60)
        at akka.typed.AskPattern$.ask(Ask.scala:64)
        at akka.typed.AskPattern$Askable$.$qmark$extension(Ask.scala:38)
        at com.example.ask.TypedAsk$$anonfun$root$1$$anonfun$apply$2.apply(TypedAsk.scala:26)
        at com.example.ask.TypedAsk$$anonfun$root$1$$anonfun$apply$2.apply(TypedAsk.scala:22)
        at akka.typed.ScalaDSL$Static.message(ScalaDSL.scala:240)
        at akka.typed.ActorAdapter$$anonfun$receive$1.applyOrElse(Impl.scala:31)
        at akka.actor.Actor$class.aroundReceive(Actor.scala:480)
        at akka.typed.ActorAdapter.aroundReceive(Impl.scala:17)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:526)
        at akka.actor.ActorCell.invoke(ActorCell.scala:495)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
        at akka.dispatch.Mailbox.run(Mailbox.scala:224)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)

^C
```

**Comment by durban** [comment]
Note: this seems to be fixed in 2.4.11.

**Comment by durban** [comment]
I was wrong, it is _not_ fixed. The tuples were modified, so that now on both sides there is a 3-tuple. However, in the failure case, the match still fails, since the third item is `null`, and the pattern `promiseRef: PromiseActorRef` doesn't match `null`.
